fuzzers/eclipser/fuzzer.py

```diff
@@ -11,88 +11,181 @@
-    encoded_temp_corpus = os.path.join(os.path.dirname(input_corpus),
-                                       'temp-corpus')
-    if not os.path.exists(encoded_temp_corpus):
-        os.mkdir(encoded_temp_corpus)
 
-    print('[fuzz] Running target with Eclipser')
+def eclipser(input_corpus, output_corpus, target_binary):
+    """Run Eclipser."""
+    # We will use output_corpus as a directory where AFL and Eclipser sync their
+    # test cases with each other. For Eclipser, we should explicitly specify an
+    # output directory under this sync directory.
+    eclipser_out = os.path.join(output_corpus, "eclipser_output")
     command = [
         'dotnet',
         '/Eclipser/build/Eclipser.dll',
-        'fuzz',
         '-p',
         target_binary,
-        '-t',
-        '1048576',  # FIXME: Find the max value allowed here.
+        '-s',
+        output_corpus,
         '-o',
-        encoded_temp_corpus,
-        '--src',
-        'file',
-        '--initarg',
-        'foo',  # Specifies how command line argument is passed, just a file.
-        '-f',
+        eclipser_out,
+        '--arg',  # Specifies the command-line of the program.
+        'foo',
+        '-f',  # Specifies the path of file input to fuzz.
         'foo',
-        '--maxfilelen',
-        # Default is too low (8 bytes), match experiment config at:
-        # https://github.com/SoftSec-KAIST/Eclipser-Artifact/blob/6aadf02eeadb0416bd4c5edeafc8627bc24ebc82/docker-scripts/experiment-scripts/package-exp/run_eclipser.sh#L25
-        '1048576',
-        # Default is low (0.5 sec), recommended to use higher:
-        # https://github.com/google/fuzzbench/issues/70#issuecomment-596060572
+        '-v',  # Controls the verbosity.
+        '2',
         '--exectimeout',
         '5000',
     ]
-    if os.listdir(input_corpus):  # Important, otherwise Eclipser crashes.
+    if os.listdir(input_corpus):  # Specify inputs only if any seed exists.
         command += ['-i', input_corpus]
-    print('[fuzz] Running command: ' + ' '.join(command))
+    print('[eclipser] Run Eclipser with command: ' + ' '.join(command))
     subprocess.Popen(command)
 
-    process = Process(target=copy_corpus_directory,
-                      args=(
-                          encoded_temp_corpus,
-                          output_corpus,
-                      ))
-    process.start()
-
-
-def copy_corpus_directory(encoded_temp_corpus, output_corpus):
-    """Copies corpus periodically from encoded corpus directory into output
-  directory."""
-    while True:
-        # Wait for initial fuzzer initialization, and after every copy.
-        time.sleep(120)
-
-        subprocess.check_call([
-            'dotnet',
-            '/Eclipser/build/Eclipser.dll',
-            'decode',
-            '-i',
-            os.path.join(encoded_temp_corpus, 'testcase'),
-            '-o',
-            output_corpus,
-        ])
+
+def afl_master(input_corpus, output_corpus, target_binary):
```

Comment:
Can you please use afl_fuzzer.run_afl_fuzz to avoid code duplication with additional_flags option (that is also important as we sometimes update the helper for corner cases).

Reply:
I initially tried to reuse afl_fuzzer.run_afl_fuzz(), but it turned out that '-d' option in run_afl_fuzz() is incompatible with '-M' or '-S' options. When such options are given together, AFL refuses to run (cf. https://github.com/google/AFL/blob/fab1ca5ed7e3552833a18fc2116d33a9241699bc/afl-fuzz.c#L7510). 

So at the moment, it seems impossible to reuse run_afl_fuzz() here. Removing '-d' option from run_afl_fuzz() is one possible option, but it will affect many other fuzzers relying on this function. So I thought it would be better to address it in a separate issue or PR.

Reply:
Basically, add a condition in run_afl_fuzzer, like:

# '-d' flag is incompatible with '-M' and '-S' options.
if not additional_flags or ('-M' not in additional_flags and '-S' not in additional_flags):
  flags.append('-d') 

We shouldn't be duplicating code, esp afl one we keep improving from time to time.

Reply:
I see, that seems to be a nicer solution. I will add a new commit to resolve this.

Reply:
@jonathanmetzman While refactoring fuzzers/afl/fuzzer.py, I found a confusing comment.  In commit 60b67bf7, the comment says `Use deterministic mode as it does best when we don't have seeds which is often the case`, but isn't `-d` option intended to "skip" deterministic mode?